src/editor/action/action_label.cpp

```diff
@@ -88,4 +88,42 @@ void editor_action_label_delete::perform_without_undo(map_context& mc) const
 	mc.get_labels().set_label(loc_, "");
 }
 
+IMPLEMENT_ACTION(label_replace)
+
+editor_action* editor_action_label_replace::perform(map_context& mc) const
+{
+	editor_action_ptr undo(new editor_action_label_replace(new_loc_, loc_));
+
+	perform_without_undo(mc);
+	return undo.release();
```

Comment:
A task for a separate PR, but this method should be returning the std::unique_ptr instead of returning an owning raw pointer.